### DISCUSSION TITLE
Need to Defined Download Directory for Selenium-Wire Driver

### DISCUSSION THREAD

**Discussion by Jonathynlee** [seleniumbase/SeleniumBase#2437] [General]
Hello, 
I am using the wire=True option for the raw driver on lambda. 
The lambda is preventing download required by sw. Is there a way similar to seleniumwire_options that I can change this to the tmp directory?

Thanks!

**Comment by mdmintz** [comment]
Duplicate of https://github.com/seleniumbase/SeleniumBase/issues/1557#issuecomment-1287636878

--------

SeleniumBase creates the ``downloaded_files`` folder in the working directory where ``pytest`` is invoked. Any click-initiated downloads will go there. It's also used for any special files needed. It also holds lock files to prevent issues with multi-threading. The folder resets at the start of every new ``pytest`` run so that past test runs don't interfere with new ones. The folder is hard-coded there to prevent issues. And since the folder is reset at the start of new ``pytest`` runs, you wouldn't want to use any other existing folders for it.

There are lots of built-in test methods that are specially made for that folder, such as:

```python
self.get_downloads_folder()

self.get_browser_downloads_folder()

self.get_path_of_downloaded_file(file)

self.is_downloaded_file_present(file)

self.delete_downloaded_file_if_present(file)
# Duplicates: self.delete_downloaded_file(file)

self.assert_downloaded_file(file)

self.get_downloaded_files(regex=None, browser=False)

self.get_data_from_downloaded_file(file, timeout=None, browser=False)

self.assert_data_in_downloaded_file(data, file, timeout=None, browser=False)
```

If you need files in a different folder, use Python ``os``, ``sys``, and ``shutil`` libraries for copying the files you need into a different folder after they get downloaded to the ``downloaded_files`` folder.

There's also a `pytest` command-line option to archive existing downloads instead of deleting them:
https://github.com/seleniumbase/SeleniumBase/blob/570910c6f6e079cddd00c23e826e202c6489f39e/README.md?plain=1#L673

(A copy of those downloads will go into a folder in the `archived_files/` folder at the end of the `pytest` run.)

**Reply by Jonathynlee** [reply]
I am unable to download and move the file. The folder it is attempting to download to is in a read only directory. 

Additionally, I am using seleniumbase raw.

Thoughts? 
Thanks!

**Reply by mdmintz** [reply]
If not using the `SB()` or `BaseCase` / `pytest` formats, then create a `downloaded_files` directory in the location where you run your scripts.
